delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/Files.scala

```diff
@@ -203,43 +206,49 @@ final class Files(
     *
     * @param id        the file identifier to expand as the iri of the file
     * @param project   the project where the storage belongs
+    * @param headers   the HTTP headers to decide whether or not the content can be fetched (using its accept headers)
     */
   def fetchContent(
       id: IdSegment,
-      project: ProjectRef
-  )(implicit caller: Caller): IO[FileRejection, AkkaSource] =
-    fetchContent(id, project, None).named("fetchFileContent", moduleType)
+      project: ProjectRef,
+      headers: Seq[HttpHeader] = Seq.empty
```

Comment:
We could check headers in the route no so as not to handle this in this layer ?
And if at one point we fetch files from another plugin for example, http headers won't be relevant.

Reply:
Yes we could, but that made the actual logic quite complicated in the routes.
If they are not relevant is not important, that's why they have a default value

Reply:
I've just added methods to handle the case of calling files.fetch from somewhere outside of the routes where headers are not important.